### PR TITLE
Activate Logic in Sidebar Buttons

### LINKED ISSUE #57: Buy and Sell on Sidebar
**Is your feature request related to a problem? Please describe.**
Getting the buttons functional

**Describe the solution you'd like**
Put the proper functions connected to the buy and sell button.

### PR DESCRIPTION
**Pull request by krferrell**
## Changes
_List Changes Introduced by this PR_
1. Enables buy and sell features on the sidebar
2. Renders info when user clicks more information button

## Purpose
This functionality allows users to buy and sell a pokemon from the sidebar. Also, it allows the user to get more information on the pokemon they are interested in. 

## Approach
Lifted necessary state and functions, then passes it to the components that need it. 

Closes #57

### CODE REVIEW

File: pokeapp/src/Brain.js [hunk 1]
```diff
@@ -130,15 +136,57 @@ export const Brain = () => {
...
+        tempPokeList[pokeIndex].quantity--;
+      } else if (tempPokeList[pokeIndex].quantity === 1) {
+        tempPokeList[pokeIndex].quantity--;
+        tempPokeList = tempPokeList.filter(
+          (pokemon) => pokemon.id !== parseInt(e.target.value)
+        );
+        setMyPokeList(tempPokeList);
+      }
+    }
+  };
```
Comment (kmausolf) [inline pokeapp/src/Brain.js:173]: Might be a good idea to keep `myPokelist` in Redux, then add this function and the `addToPokeList` function as reducers.  `myPokelist` should live in global state.

File: pokeapp/src/Brain.js [hunk 2]
```diff
@@ -130,15 +136,57 @@ export const Brain = () => {
...
+
+  const subtractPokeDollars = (pokemon) => {
+    setPokeDollars((currentAmount) => currentAmount - pokemon.value);
+  };
+
+  const addPokeDollars = (pokemon) => {
+    setPokeDollars((currentAmount) => currentAmount + pokemon.value);
+  };
```
Comment (kmausolf) [inline pokeapp/src/Brain.js:181]: This is also global state.  We should store it in Redux rather than having one component manage the whole app.

**Review by gonzalespaulb — APPROVED** [review]
LGTM

**Review by JayDampitan — APPROVED** [review]
Looks good!

**Pull request merged**